tests/NLog.UnitTests/LogFactoryTests.cs

```diff
@@ -112,6 +109,38 @@ public void InvalidXMLConfiguration_ThrowErrorWhen_ThrowExceptionFlagIsSet()
         }
 
         [Fact]
+        public void SecondaryLogFactoryDoesNotTakePrimaryLogFactoryLock()
```

Comment:
Yes. We run the thread while holding the lock of the global `LogFactory`. So if the code in the thread takes the same lock, we will deadlock and the `Join()` method will time out.